manifests/configaware.go

```diff
@@ -141,39 +135,22 @@ func (frs *fileResourceStore) SetWorkloadContainerImage(ctx context.Context, res
 		return ErrResourceNotFound(resourceID.String())
 	}
 	if resWithOrigin.configFile == nil {
-		if err := frs.setManifestWorkloadContainerImage(resWithOrigin.resource, container, newImageID); err != nil {
+		if err := ca.rawfiles.setManifestWorkloadContainerImage(resWithOrigin.resource, container, newImageID); err != nil {
 			return err
 		}
-	} else if err := frs.setConfigFileWorkloadContainerImage(ctx, resWithOrigin.configFile, resWithOrigin.resource, container, newImageID); err != nil {
+	} else if err := ca.setConfigFileWorkloadContainerImage(ctx, resWithOrigin.configFile, resWithOrigin.resource, container, newImageID); err != nil {
 		return err
 	}
 	// Reset resources, since we have modified one
-	frs.resetResources()
+	ca.resetResources()
 	return nil
 }
 
-func (frs *fileResourceStore) setManifestWorkloadContainerImage(r resource.Resource, container string, newImageID image.Ref) error {
-	fullFilePath := filepath.Join(frs.baseDir, r.Source())
-	def, err := ioutil.ReadFile(fullFilePath)
-	if err != nil {
-		return err
-	}
-	newDef, err := frs.manifests.SetWorkloadContainerImage(def, r.ResourceID(), container, newImageID)
-	if err != nil {
-		return err
-	}
-	fi, err := os.Stat(fullFilePath)
-	if err != nil {
-		return err
-	}
-	return ioutil.WriteFile(fullFilePath, newDef, fi.Mode())
-}
-
-func (frs *fileResourceStore) setConfigFileWorkloadContainerImage(ctx context.Context, cf *ConfigFile, r resource.Resource,
+func (ca *configAware) setConfigFileWorkloadContainerImage(ctx context.Context, cf *ConfigFile, r resource.Resource,
 	container string, newImageID image.Ref) error {
 	if cf.PatchUpdated != nil {
-		return frs.updatePatchFile(ctx, cf, func(previousManifests []byte) ([]byte, error) {
-			return frs.manifests.SetWorkloadContainerImage(previousManifests, r.ResourceID(), container, newImageID)
+		return ca.updatePatchFile(ctx, cf, func(previousManifests []byte) ([]byte, error) {
+			return ca.rawfiles.manifests.SetWorkloadContainerImage(previousManifests, r.ResourceID(), container, newImageID)
```

Comment:
Shouldn't we be invoking `SetWorkloadContainerImage()` directly on rawfiles? Otherwise I think we should keep our own manifests copy.

Reply:
We can't call `SetWorkloadContainerImage()` since it operates on files. I would keep a copy of the manifests then.

Reply:
Yeah, `rawfiles` is doing double duty, by being the implementation for rawfiles; and (for configAware) being a holder for some stuff it needs, like the manifests. I'm not too nervous about this -- is the objection about maintaining encapsulation (rather that correctness)?

Reply:
Yes, encapsulation and clear resource ownership.

Reply:
OK, it's a fair objection -- I'm trying to get away with too much :-) I'll fix it up.